code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -86,24 +86,38 @@
 	R.updateicon()
 
 /obj/item/borg/upgrade/rename
-	name = "robot reclassification board"
-	desc = "Used to rename a cyborg."
+	var/heldname = ""
+	name = "robot rename board"
+	desc = "Used to rename a cyborg, or allow a cyborg to rename themselves."
 	icon_state = "cyborg_upgrade1"
-	var/heldname = "default name"
 
 /obj/item/borg/upgrade/rename/attack_self(mob/user as mob)
-	heldname = stripped_input(user, "Enter new robot name", "Robot Reclassification", heldname, MAX_NAME_LEN)
+	heldname = stripped_input(user, "Enter new robot name to force, or leave clear to let the robot pick a name", "Robot Rename", heldname, MAX_NAME_LEN)
```

Comment:
`as null|anything`? cant remember if that's the correct syntax
but you need to do more than just add a clause, because, for instance, player can set borg name to a single space